src/test/java/TestSuite.java

```diff
@@ -0,0 +1,199 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.junit.platform.suite.api.IncludeTags;
+import org.junit.platform.suite.api.SelectClasses;
+import org.junit.platform.suite.api.Suite;
+import parser.JsonParser;
+import parser.NoSuchFileException;
+import parser.Parser;
+import shop.Cart;
+import shop.RealItem;
+import shop.VirtualItem;
+
+import java.io.File;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+@Suite
+@SelectClasses({JsonParserTest.class, RealItemTest.class, VirtualItemTest.class, CartTest.class})
+@IncludeTags({"JsonParserTest", "RealItemTest", "VirtualItemTest", "CartTest"})
+class TestSuite {
+}
+
+@Tag("JsonParserTest")
+@DisplayName("JsonParser Class Tests")
+class JsonParserTest {
+    Parser parser;
+    File file;
+    Cart cart;
+
+    @BeforeEach
+    public void setup() {
+        parser = new JsonParser();
+        cart = new Cart("test-cart");
+        file = new File("src/main/resources/" + cart.getCartName() + ".json");
+    }
+
+    @AfterEach
+    public void cleanUp() {
+        file.deleteOnExit();
+        cart = null;
+    }
+
+    @DisplayName("Should Write To File")
+    @Test
+    public void shouldWriteToFile() {
+        parser.writeToFile(cart);
+        // checking that file is not empty
+        assertNotEquals(0, file.length());
+    }
+
+    @DisplayName("Should Read From File")
+    @ParameterizedTest
+    @ValueSource(strings = {"andrew-cart", "eugen-cart"})
+    public void shouldReadFromFile(String cartName) {
+        cart = parser.readFromFile(new File("src/main/resources/" + cartName + ".json"));
+        // checking that cart is not null
+        assertNotNull(cart);
+    }
+
+    // NoSuchFileException Test with 5 datasets
+    @DisplayName("Should Throw NoSuchFileException")
+    @ParameterizedTest
+    @ValueSource(strings = {"", "file", "file.json", "file.txt", " file "})
+    public void shouldThrowNoSuchFileException(String fileName) {
+        assertThrows(NoSuchFileException.class,
+                () -> parser.readFromFile(new File("src/main/resources/" + fileName)));
+    }
+
+    // Testing Exception Message using nested group assertion
+    // Decided to disable this test
+    @Disabled
+    @DisplayName("Exception Message Test")
+    @Test
+    public void shouldMatchExceptionMessage() {
+        String fileName = "dummy";
+        String expectedErrorMessage = "File " + fileName + ".json not found!";
+        assertAll("TestExceptionThrown",
+                () -> {
+                    Exception exception = assertThrows(NoSuchFileException.class,
+                            () -> parser.readFromFile(new File(fileName)));
+
+                    assertAll("TestExceptionMessage",
+                            () -> assertEquals(expectedErrorMessage, exception.getMessage())
+                    );
+                }
+        );
+    }
```

Comment:
If test enabled, is it passed?

Reply:
Yes, all tests are passed. 

![tests](https://user-images.githubusercontent.com/117095031/207594340-9f88eab7-2bab-45ea-8003-c60c460166c6.png)

As you can see, when enabling the previously ignored test, it is passed along with the rest of the tests